packages/rs-drive-abci/src/abci/app/consensus.rs

```diff
@@ -17,16 +19,19 @@ use tenderdash_abci::proto::abci as proto;
 pub struct ConsensusAbciApplication<'a, C> {
     /// Platform
     platform: &'a Platform<C>,
-    /// The current transaction
-    transaction: RwLock<Option<Transaction<'a>>>,
+    /// The current GroveDb transaction
+    transaction: RefCell<Option<Transaction<'a>>>,
+    /// The current block execution context
+    block_execution_context: RefCell<Option<BlockExecutionContext>>,
```

Comment:
We should not use RefCell, I also do not know why you are putting this here.

Reply:
Reverted to RwLock. Explained verbally why I moved BlockExecutionContext here.